pkg/planner/core/rule_join_reorder.go

```diff
@@ -401,74 +403,175 @@ type baseSingleGroupJoinOrderSolver struct {
 	*basicJoinGroupInfo
 }
 
-func (s *baseSingleGroupJoinOrderSolver) generateLeadingJoinGroup(curJoinGroup []base.LogicalPlan, hintInfo *h.PlanHints, hasOuterJoin bool) (bool, []base.LogicalPlan) {
-	var leadingJoinGroup []base.LogicalPlan
+// generateLeadingJoinGroup processes both flat and nested leading hints through the unified LeadingList structure
+func (s *baseSingleGroupJoinOrderSolver) generateLeadingJoinGroup(
+	curJoinGroup []base.LogicalPlan, hintInfo *h.PlanHints, hasOuterJoin bool,
+) (bool, []base.LogicalPlan) {
+	if hintInfo == nil || hintInfo.LeadingList == nil {
+		return false, nil
+	}
+	// Use the unified nested processing for both flat and nested structures
+	return s.generateNestedLeadingJoinGroup(curJoinGroup, hintInfo.LeadingList, hasOuterJoin)
+}
+
+// generateNestedLeadingJoinGroup processes both flat and nested LEADING hint structures
+func (s *baseSingleGroupJoinOrderSolver) generateNestedLeadingJoinGroup(
+	curJoinGroup []base.LogicalPlan, leadingList *ast.LeadingList, hasOuterJoin bool,
+) (bool, []base.LogicalPlan) {
 	leftJoinGroup := make([]base.LogicalPlan, len(curJoinGroup))
 	copy(leftJoinGroup, curJoinGroup)
+
+	leadingJoin, remainingGroup, ok := s.buildLeadingTreeFromList(leadingList, leftJoinGroup, hasOuterJoin)
+	if !ok {
+		return false, nil
+	}
+	s.leadingJoinGroup = leadingJoin
+	return true, remainingGroup
+}
+
+// buildLeadingTreeFromList recursively constructs a LEADING tree, with items coming from ast.LeadingList.Items (*ast.HintTable | *ast.LeadingList)
+func (s *baseSingleGroupJoinOrderSolver) buildLeadingTreeFromList(
```

Comment:
Please add more comments to explain how `leadingList *ast.LeadingList` looks like.
For example, if the hint is `leading(t1, (t2, t3), (t4, (t5, t6, t7)))`, how this structure would look like.